From d2c68aa27cd88527c0700616ad767e6094dcaf28 Mon Sep 17 00:00:00 2001
From: root <root@example.org>
Date: Sat, 23 Apr 2022 14:03:47 +0000
Subject: [PATCH] GCC: fix ambiguous references in QueryProcessor

GCC does not allow aliases with same name. Use explicit namespace to
avoid confusion.
---
 .../internal/execution/processing/query_processor.h         | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/components/segmentation_platform/internal/execution/processing/query_processor.h b/components/segmentation_platform/internal/execution/processing/query_processor.h
index 403446b..dc3366a 100644
--- a/components/segmentation_platform/internal/execution/processing/query_processor.h
+++ b/components/segmentation_platform/internal/execution/processing/query_processor.h
@@ -22,9 +22,9 @@ class QueryProcessor {
  public:
   // TODO(ssid): Remove these indirections and replace all uses with the global
   // types.
-  using Tensor = Tensor;
-  using IndexedTensors = IndexedTensors;
-  using FeatureIndex = FeatureIndex;
+  using Tensor = segmentation_platform::processing::Tensor;
+  using IndexedTensors = segmentation_platform::processing::IndexedTensors;
+  using FeatureIndex = segmentation_platform::processing::FeatureIndex;
 
   // TODO(haileywang): Maybe use a unique_ptr<> here.
   using QueryProcessorCallback =
-- 
2.35.1

